### DISCUSSION TITLE
Use of Repeatable inside Relationship

### DISCUSSION THREAD

**Discussion by lieto-dk** [Laravel-Backpack/community-forum#353] [Q&A (Help)] [answered]
Hi, 

I have an issue while using the Repeatable field inside a Relationship field. When cliciking on the Add item for the Repeatable a new record is shown for the Relationship. My code is as below for the relationship field. Anyone with any ideas what goes wrong ?

$this->crud->addField([
            'name' => "categorylocale",
            'type' => 'relationship',
            'label' => 'Category Locales per Site',
            'tab'             => 'Locales',
            'new_item_label'  => 'Add Locale for Category',
            'subfields'   => [
                [
                    'name'  => 'site_id',
                    'type'  => 'select2',
                    'label' => 'Site',
                    'attribute' => 'country',
                    'wrapper'   => ['class' => 'form-group col-md-3'],
                ],
                [
                    'name'  => 'name_locale',
                    'type'  => 'text',
                    'label' => 'Name (locale)',
                    'wrapper'   => ['class' => 'form-group col-md-4'],
                ],
                [
                    'name'  => 'slug',
                    'type'  => 'text',
                    'label' => 'Slug',
                    'wrapper'   => ['class' => 'form-group col-md-4'],
                ],
                [
                    'name'  => 'price_limit',
                    'type'  => 'number',
                    'label' => 'Price limit',
                    'wrapper'   => ['class' => 'form-group col-md-3'],
                ],
                [
                    'name'  => 'min_price',
                    'type'  => 'number',
                    'label' => 'Min price',
                    'wrapper'   => ['class' => 'form-group col-md-3'],
                ],
                [
                    'name'  => 'max_savings_pct',
                    'type'  => 'number',
                    'label' => 'Max savings pct',
                    'wrapper'   => ['class' => 'form-group col-md-3'],
                ],
                [
                    'name'  => 'count_offers',
                    'type'  => 'number',
                    'label' => 'Offers',
                    'wrapper'   => ['class' => 'form-group col-md-3'],
                ],
                [
                    'name'  => 'page_title',
                    'type'  => 'text',
                    'label' => 'Page title',
                    'wrapper'   => ['class' => 'form-group col-md-12'],
                ],
                [
                    'name'  => 'content_title',
                    'type'  => 'text',
                    'label' => 'Content title',
                    'wrapper'   => ['class' => 'form-group col-md-12'],
                ],
                [
                    'name'  => 'description',
                    'type'  => 'summernote',
                    'label' => 'Content',
                    'wrapper'   => ['class' => 'form-group col-md-12'],
                ],
                [
                    'name'  => 'faqs',
                    'label' => 'FAQs',
                    'type'  => 'repeatable',
                    'fields' => [
                        [
                            'name'    => 'question',
                            'label'     => 'Question',
                            'type'    => 'textarea',
                            'wrapper' => ['class' => 'form-group col-md-12'],
                        ],
                        [
                            'name'    => 'answer',
                            'label'     => 'Answer',
                            'type'    => 'summernote',
                            'wrapper' => ['class' => 'form-group col-md-12'],
                        ],
                        [
                            'name'    => 'url',
                            'label'     => 'Url',
                            'type'    => 'text',
                            'wrapper' => ['class' => 'form-group col-md-12'],
                        ],
                        [
                            'name'    => 'link_text',
                            'label'     => 'Link Text',
                            'type'    => 'text',
                            'wrapper' => ['class' => 'form-group col-md-12'],
                        ],
                        [
                            'name'    => 'affiliate_id',
                            'label'     => 'Affiliate',
                            'type'      => 'select2',
                            //'entity'    => 'affiliate',
                            'model'     => "App\Models\Affiliate",
                            'attribute' => 'name',
                            'wrapper' => ['class' => 'form-group col-md-6'],
                        ],
                        [
                            'name'    => 'sorting',
                            'label'     => 'Sorting',
                            'type'    => 'text',
                            'default'     => '1',
                        ],
                    ],
                    'new_item_label'  => 'Add FAQ',
                    'init_rows' => 0,
                ],
            ],

**Comment by lieto-dk** [comment]
Have updated Backpack (pro to 1.5.0) and published assets. Tested in browser where cache was cleared. Same issue.

**Comment by pxpm** [comment, marked as answer]
Hello @lieto-dk 

At the present moment it's not possible to use a `repeatable` inside `repeatable`.  When you set the field type to `relationship` with `subfields`, under the hood you are creating a `repeatable` field. And that would fail when you add a repeatable in the subfields. 

Depending on the use-case, sometimes the `table` field type would be enough for the job. 
I have plans to make repeatables work inside other repeatables in future (prob. after v6 launch) but apart from the technical aspect, I think most difficult to achieve is the design/presentation aspect. 

I've added a section to the docs to make it more clear for other developers too. https://github.com/Laravel-Backpack/docs/pull/398

Let's see what we can come up after v6 is launched. 🙏 

Thanks @lieto-dk

**Reply by lieto-dk** [reply]
Thanks for the reply.
The table field will not work for this use case.
Will try to solve the issue with a morphMany instead.
